tests/unit/test_orchestrator.py

```diff
@@ -0,0 +1,208 @@
+"""Test the orchestrator.py module."""
```

Comment:
Since there's quite a lot of comments on the code itself, I'll delay reviewing tests until later.